lib/xcodeproj/project/object/build_configuration.rb

```diff
@@ -82,6 +82,7 @@ def type
         def resolve_build_setting(key)
           setting = build_settings[key]
           config_setting = base_configuration_reference && config[key]
+          config_setting = resolve_variable_substitution(config_setting) if !config_setting.nil? && config_setting.is_a?(String)
```

Comment:
xcconfig file parsing goes through `XCBuildConfiguration#normalize_array_settings` and we get array values from there. I'm completely ignoring those, as I believe it's mostly used for `OTHER_LDFLAGS`, but I don't fully understand why those have to be arrays, I assume this is expected in other parts of system.